Add 2.4 release notes (#860)
kyma-project/website: content/blog-posts/2022-06-30-release-notes-2.4/index.md

> <!-- overview -->

Reviewer: Please add the overview content above this line.

Author: Done.

See the overview of all changes in this release:

> - [Application Connectivity](#application-connectivity) - Application connectivity adjusted to Service Catalog removal, Application Gateway support for mTLS-OAuth

Reviewer: ```suggestion
- [Application Connectivity](#application-connectivity) - Application Gateway support for mTLS-OAuth
```

- [Eventing](#eventing) - Introduced NATS Jetstream as the default Eventing Backend 
- [Observability](#observability) - Kiali upgrade to 1.51.1
- [Serverless](#serverless) - Added support to NodeJS16, improved scheduling of Function build jobs
- [Service Management](#service-management) - Service Catalog removed
- [Service Mesh](#service-mesh) - Istio upgraded to 1.14.1

 
## Application Connectivity 
 

> ### Application Connectivity adjusted to Service Catalog removal 

Reviewer: This note is a duplication of the one added below in the SC removal section

Kyma 2.4 [no longer contains Service Catalog deployments](#service-catalog-removed). To keep up with that change in Application Connector, we removed some components that interacted with Service Catalog. This means that Application Operator and Application Broker are no longer installed on new clusters.  
This also means that on a fresh cluster, or after executing the migration script we provide, the old Application flow will no longer be present, and you will need to use the new way of connecting the external Systems.  
Note that executing the migration script is optional, and if you would like to keep the old components on the cluster, you're free to do so. 

### Application Gateway support for mTLS-OAuth 

> We introduced support for mTLS OAuth-secured APIs in Application Gateway. You can now configure your Application CRs so that Application Gateway fetches the OAuth token with the help of a TLS certificate as a credential. To learn how to do that, see Register an OAuth 2.0 mTLS-secured API. 

Reviewer: ```suggestion
We introduced support for [mTLS OAuth](https://datatracker.ietf.org/doc/html/rfc8705)-secured APIs in Application Gateway. You can now configure your Application CRs so that Application Gateway fetches the OAuth token with the help of a TLS certificate as a credential. To learn how to do that, see [Register an OAuth 2.0 mTLS-secured API](https://kyma-project.io/docs/kyma/main/03-tutorials/00-application-connectivity/ac-04-register-secured-api/#register-an-o-auth-2-0-m-tls-secured-api). 
```

## Eventing 

### Introduced NATS Jetstream as the default Eventing Backend 

> With Kyma 2.4, we changed the NATS eventing backend to use `NATS Jetstream` instead of just `Core NATS`. This updated backend improves the delivery guarantees from previously AT-MOST-ONCE to now AT-LEAST-ONCE. To facilitate this change, the NATS cluster now relies on backing storage by default.  

Reviewer: ```suggestion
With Kyma 2.4, we changed the NATS eventing backend to use [`NATS Jetstream`](https://docs.nats.io/nats-concepts/jetstream) instead of just [`Core NATS`](https://docs.nats.io/nats-concepts/core-nats). This updated backend improves the delivery guaranteed from previously AT-MOST-ONCE to now AT-LEAST-ONCE. To facilitate this change, the NATS cluster now relies on backing storage by default.  
```

Reviewer: Not sure about that but it feels like `guaranteed` was meant here 🤔

## Observability 

> To ensure compatibility after to the upgrade of Istio to 1.14, we upgraded Kiali to 1.51.1. It contains name changes of most Kiali resources; so we added a clean-up script to delete old resources after the upgrade. 

Reviewer: ```suggestion
To ensure compatibility after the Istio upgrade to v1.14, we upgraded Kiali to 1.51.1. It contains name changes of most Kiali resources, so we added a clean-up script to delete old resources after the upgrade. 
```

Reviewer: Where's that script?

 
## Serverless 
 

> ### Added support to NodeJS16 

Reviewer: ```suggestion
### Added support to NodeJS 16 
```

> With Kym 2.4, you have a new serverless runtime available – nodejs16. 

Reviewer: ```suggestion
With Kym 2.4, you have a new Serverless runtime available – NodeJS 16. 
```

This new runtime not only brings a new version of nodejs, but also new versions of  bundled OpenTelemetry SDK. All this enhances the development experience for nodejs developers and improves traceability of the requests handled by your nodejs functions. 
 
### Improved scheduling of Function build jobs 

> We have fixed the resource configuration for some of the Kyma components to reduce their cpu overcommitment. Additionally, we have removed the fixed requirements for computation resources for Function build jobs. With those changes, Function build jobs have better conditions to be scheduled, and your Functions won't be stuck in building phase. 

Reviewer: ```suggestion
We have fixed the resource configuration for some of the Kyma components to reduce their CPU overcommitment. Additionally, we have removed the fixed requirements for computation resources for Function build jobs. With those changes, Function build jobs have better conditions to be scheduled, and your Functions won't be stuck in the building phase. 
```

- Rafter 

Note that your Service Catalog resources will not be migrated to any other solution. As mentioned in the [Service Catalog deprecation update](https://kyma-project.io/blog/2021/12/7/release-notes-20#service-catalog-deprecation-update), we recommend you to use [service operators for Service Management in Kyma](https://kyma-project.io/docs/kyma/main/01-overview/main-areas/service-management/smgt-01-overview/). 

 
If you already switched to another solution and want to remove the obsolete CRDs from your cluster, run the cleanup script provided in the [Migration Guide](https://kyma-project.io/docs/kyma/2.4/migration-guide-2.3-2.4) **after** you upgrade from Kyma 2.3 to 2.4. 
 
 Service Catalog removal also affects Application Connectivity. We removed some components that interacted with Service Catalog, which means that Application Operator and Application Broker are no longer installed on new clusters. This also means that on fresh clusters, or after executing the migration script we provide, the old Application flow will no longer be present, and you will need to use the new way of connecting the external Systems. 
## Service Mesh 

### Istio upgraded to 1.14.1 

> In this release, we upgraded Istio from 1.13.2 to 1.14.1. For more details on the changes, read the official [Istio 1.14.1 release notes](https://istio.io/latest/news/releases/1.14.x/announcing-1.14.1/). 

Reviewer: ```suggestion
In this release, we upgraded Istio from 1.13.2 to 1.14.1. For more details, read the official [Istio 1.14.1 release notes](https://istio.io/latest/news/releases/1.14.x/announcing-1.14.1/). 
```

<!-- overview -->

See the overview of all changes in this release:

- [General](#general) - Removal of Rafter
- [Application Connectivity](#application-connectivity) - Application connectivity adjusted to Service Catalog removal, Application Gateway support for mTLS-OAuth
- [Eventing](#eventing) - Introduced NATS Jetstream as the default Eventing Backend 
- [Observability](#observability) - Kiali upgrade to 1.51.1
- [Serverless](#serverless) - Added support to NodeJS16, improved scheduling of Function build jobs
- [Service Management](#service-management) - Service Catalog removed
- [Service Mesh](#service-mesh) - Istio upgraded to 1.14.1

## General 
 

> ### Rafter removed from Kyma components 

Reviewer: I would remove this note. We mention it already in the `Service Catalog removal` section. 
cc: @PK85 @m00g3n

 
With Kyma 2.4, we removed Rafter from the list of installed components. Rafter handles asset storage and is used internally by Helm Broker and Runtime Agent. 
When you upgrade from Kyma 2.3 to 2.4, Rafter is not removed automatically - you can continue using it if you prefer. If you dont need Rafter (or any of the removed components), refer to the [migration guide](ADD LINK HERE) that describes how to get rid of them. 

## Application Connectivity 
 

> ### Application Connectivity adjusted to Service Catalog removal 

Reviewer: After all, I'd combine this note with the one regarding Service Catalog removal. See my suggestion in the `Service Catalog removal` section.

## Service Management 

### Service Catalog removed 

With Kyma 1.23, we announced the [deprecation of Service Catalog](https://kyma-project.io/blog/2021/6/2/release-notes-123#service-management). Now in Kyma 2.4, we completely removed it. That's why in the 2.4 release, the following Service Catalog-related components are no longer installed as part of Kyma: 

- Service Catalog 

- Helm Broker 

- Application Broker 

- Application Operator 

- Service Binding Usage Controller 

- Rafter 

Note that your Service Catalog resources will not be migrated to any other solution. As mentioned in the [Service Catalog deprecation update](https://kyma-project.io/blog/2021/12/7/release-notes-20#service-catalog-deprecation-update), we recommend you to use [service operators for Service Management in Kyma](https://kyma-project.io/docs/kyma/main/01-overview/main-areas/service-management/smgt-01-overview/). 

 

> If you already switched to another solution and want to remove the obsolete CRDs from your cluster, run the cleanup script provided in the [Migration Guide](link) **after** you upgrade from Kyma 2.3 to 2.4. 

Reviewer: ```suggestion
If you already switched to another solution and want to remove the obsolete CRDs from your cluster, run the cleanup script provided in the [Migration Guide](https://kyma-project.io/docs/kyma/2.4/migration-guide-2.3-2.4) **after** you upgrade from Kyma 2.3 to 2.4. 
```

>  

Reviewer: ```suggestion
 
 Service Catalog removal also affects Application Connectivity. We removed some components that interacted with Service Catalog, which means that Application Operator and Application Broker are no longer installed on new clusters.  This also means that on fresh clusters, or after executing the migration script we provide, the old Application flow will no longer be present, and you will need to use the new way of connecting the external Systems. 
```

Reviewer: After all, I'd combine all SC removal-related notes into one to have all the info in one place. @PK85 @janmedrek please have a look and let me know if you're okay with my suggestion 🙂

---
title: "Kyma 2.4"
author:
  name: "Varban Vasilev, PM/PO @Kyma"
tags:
  - release-notes
type: release
releaseTag: "2.4.0"
redirectFrom:
  - "/blog/release-notes-24"
---

Swooosh… wow that was fast - the first half of 2022 is over already, and we’re at Kyma release 2.4!
We hope this summery update finds you well; maybe you’re going through your wardrobe and packing for vacation right now? No matter if you’re already sipping a cocktail on the beach or busy planning your vacation, find some time to read our hot news.
To name just a few of the improvements and updates that we packed into this release, Eventing got a new backend and Serverless supports a new runtime.
We’ve also finalized our spring cleaning and completely removed Service Catalog and Rafter. The Application Connectivity and Observability components have been adjusted accordingly.

<!-- overview -->

See the overview of all changes in this release:

- [Application Connectivity](#application-connectivity) - Application Gateway support for mTLS-OAuth
- [Eventing](#eventing) - Introduced NATS Jetstream as the default Eventing Backend 

> - [Observability](#observability) - Kiali upgrade to 1.51.1

Reviewer 2: Moving this below the Istio section

```suggestion
```

> - [Serverless](#serverless) - Added support to NodeJS16, improved scheduling of Function build jobs

Reviewer 2: ```suggestion
- [Serverless](#serverless) - Added support to NodeJS 16, improved scheduling of Function build jobs
```

- [Service Management](#service-management) - Service Catalog removed
- [Service Mesh](#service-mesh) - Istio upgraded to 1.14.1

> 

Reviewer 2: ```suggestion
- [Observability](#observability) - Kiali upgrade to 1.51.1

```

 
## Application Connectivity 
 
### Application Gateway support for mTLS-OAuth 

We introduced support for [mTLS OAuth](https://datatracker.ietf.org/doc/html/rfc8705)-secured APIs in Application Gateway. You can now configure your Application CRs so that Application Gateway fetches the OAuth token with the help of a TLS certificate as a credential. To learn how to do that, see [Register an OAuth 2.0 mTLS-secured API](https://kyma-project.io/docs/kyma/main/03-tutorials/00-application-connectivity/ac-04-register-secured-api/#register-an-o-auth-2-0-m-tls-secured-api). 

## Eventing 

### Introduced NATS Jetstream as the default Eventing Backend 

With Kyma 2.4, we changed the NATS eventing backend to use [`NATS Jetstream`](https://docs.nats.io/nats-concepts/jetstream) instead of just [`Core NATS`](https://docs.nats.io/nats-concepts/core-nats). This updated backend improves the delivery guarantees from previously AT-MOST-ONCE to now AT-LEAST-ONCE. To facilitate this change, the NATS cluster now relies on backing storage by default.  

> ## Observability 

Reviewer 2: I'd put this section below the section about the Istio upgrade.

```suggestion
```

> To ensure compatibility after the Istio upgrade to v1.14, we upgraded Kiali to 1.51.1. It contains name changes of most Kiali resources, so we added a clean-up script to delete old resources after the upgrade. 

Reviewer 2: ```suggestion
```

- Rafter 

Note that your Service Catalog resources will not be migrated to any other solution. As mentioned in the [Service Catalog deprecation update](https://kyma-project.io/blog/2021/12/7/release-notes-20#service-catalog-deprecation-update), we recommend you to use [service operators for Service Management in Kyma](https://kyma-project.io/docs/kyma/main/01-overview/main-areas/service-management/smgt-01-overview/). 
 
If you already switched to another solution and want to remove the obsolete CRDs from your cluster, run the cleanup script provided in the [Migration Guide](https://kyma-project.io/docs/kyma/2.4/migration-guide-2.3-2.4) **after** you upgrade from Kyma 2.3 to 2.4. 
 
Service Catalog removal also affects Application Connectivity. We removed some components that interacted with Service Catalog, which means that Application Operator and Application Broker are no longer installed on new clusters. This also means that on fresh clusters, or after executing the migration script we provide, the old Application flow will no longer be present, and you will need to use the new way of connecting the external Systems. 

## Service Mesh 

### Istio upgraded to 1.14.1 

> In this release, we upgraded Istio from 1.13.2 to 1.14.1. For more details, read the official [Istio 1.14.1 release notes](https://istio.io/latest/news/releases/1.14.x/announcing-1.14.1/). 

Reviewer 2: ```suggestion
In this release, we upgraded Istio from 1.13.2 to 1.14.1. For more details, read the official [Istio 1.14.1 release notes](https://istio.io/latest/news/releases/1.14.x/announcing-1.14.1/). 

## Observability 

To ensure compatibility after the Istio upgrade to v1.14, we upgraded Kiali to 1.51.1. It contains name changes of most Kiali resources, so we added a clean-up script to delete old resources after the upgrade. 
```

---

Swooosh… wow that was fast - the first half of 2022 is over already, and we’re at Kyma release 2.4!
We hope this summery update finds you well; maybe you’re going through your wardrobe and packing for vacation right now? No matter if you’re already sipping a cocktail on the beach or busy planning your vacation, find some time to read our hot news.
To name just a few of the improvements and updates that we packed into this release, Eventing got a new backend and Serverless supports a new runtime.
We’ve also finalized our spring cleaning and completely removed Service Catalog and Rafter. The Application Connectivity and Observability components have been adjusted accordingly.

<!-- overview -->

See the overview of all changes in this release:

- [Application Connectivity](#application-connectivity) - Application Gateway support for mTLS-OAuth
- [Eventing](#eventing) - Introduced NATS Jetstream as the default Eventing Backend 
- [Serverless](#serverless) - Added support to NodeJS 16, improved scheduling of Function build jobs
- [Service Management](#service-management) - Service Catalog removed
- [Service Mesh](#service-mesh) - Istio upgraded to 1.14.1

> - [Observability](#observability) - Kiali upgrade to 1.51.1

Reviewer 2: ```suggestion
- [Observability](#observability) - Kiali upgraded to 1.51.1
```

 
If you already switched to another solution and want to remove the obsolete CRDs from your cluster, run the cleanup script provided in the [Migration Guide](https://kyma-project.io/docs/kyma/2.4/migration-guide-2.3-2.4) **after** you upgrade from Kyma 2.3 to 2.4. 
 
Service Catalog removal also affects Application Connectivity. We removed some components that interacted with Service Catalog, which means that Application Operator and Application Broker are no longer installed on new clusters. This also means that on fresh clusters, or after executing the migration script we provide, the old Application flow will no longer be present, and you will need to use the new way of connecting the external Systems. 

## Service Mesh 

### Istio upgraded to 1.14.1 

In this release, we upgraded Istio from 1.13.2 to 1.14.1. For more details, read the official [Istio 1.14.1 release notes](https://istio.io/latest/news/releases/1.14.x/announcing-1.14.1/). 

## Observability 

> 

Reviewer 2: ```suggestion

### Kiali upgraded to 1.51.1

```
